ESPixelStick/src/input/InputFPPRemotePlayFile.cpp

```diff
@@ -177,12 +177,26 @@ void c_InputFPPRemotePlayFile::GetStatus (JsonObject& JsonStatus)
 {
     // xDEBUG_START;
 
+    // NOTE:
+    // variables ending in "Total" reflect total     play time
+    // variables ending in "Rem"   reflect remaining play time
+
+    // File uses milliseconds to store elapsed and total play time
     uint32_t mseconds = FrameControl.ElapsedPlayTimeMS;
-    // BUGBUG -- can overflow, resulting in incorrect time calculation (uint32 * uint32 ==> requires uint64 result)
-    uint32_t msecondsTotal = FrameControl.FrameStepTimeMS * FrameControl.TotalNumberOfFramesInSequence;
+    uint32_t msecondsTotal;
+    if (__builtin_mul_overflow(FrameControl.FrameStepTimeMS, FrameControl.TotalNumberOfFramesInSequence, &msecondsTotal)) {
+        // returned non-zero: there has been an overflow
+        msecondsTotal = 1000u; // set to one second total when overflow occurs
```

Comment:
Just a note on the use of 1000 to convert time. I have been eliminating these as I find them and replacing them with predefined values in EspixelStick.h. Would you mind converting these as you find them?

Reply:
Also, please avoid placing implementation code in include files. Except for a simple template operation, such code should be in a CPP file.

Reply:
> Also, please avoid placing implementation code in include files. Except for a simple template operation, such code should be in a CPP file.

Hi Martin, thanks for looking over the code.  I appreciate your thoughts, and will be incorporating the feedback.  Can you help me understand which code the above comment was referencing?   While I'll look again, I don't think I had non-template implementations defined in header files, and the two comments in this review both seemed to reference `.cpp` files, not headers?

Thanks again!

Reply:
It was not so much that function but the creation of your time conversion function and its use in multiple locations. That functionality should probably be located in an ESPixelStickUtility.cpp/hpp file set (and a whole bunch of such utilities should get moved there. Any function that is in an include file and is not used in an interrupt is a candidate for that (including the json access macros). The json macros started simple but ended up getting complex. At some point they should have been converted to functions and moved out of the EspixelStick.h file.

Reply:
Ah.  I understand now.  That function **_is_** a template function.
The use of a template enables a compile-time error when the argument (the fixed-size array, which is passed-by-reference) refers to an array that is too small.

The additional runtime checks of the results of `snprintf()` are there to catch future errors, such as copy/paste of the code, without updating the critical static_assert() ... belt & suspenders approach.

I will update the PR when I've integrated your feedback... thanks again!

Reply:
Functions moved to separate header with commits 7b4bf53 and c954109.

Reduce usage of the magic numbers with commit b883525.